### DISCUSSION TITLE
Early builds of Java 21 in JBang

### DISCUSSION THREAD

**Discussion by aghasemi** [jbangdev/jbang#1626] [General]
Hi,

Sorry if it has been discussed before. Is there a way to use (for example) early Java 21 builds or some other JDK to run a JBang script?

Many thanks
Best

**Comment by maxandersen** [comment]
`jbang --java <version>`

i.e. `jbang --java 21 properties@jbangdev`

**Comment by maxandersen** [comment]
note, if you want to use some other jdk simplest is to setup java_home or java in PATH. That will be used as the default jvm.
